frontend/src/components/GTerminal.vue

```diff
@@ -0,0 +1,914 @@
+<!--
+Copyright (c) 2019 by SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+ -->
+
+<template>
```

Comment:
very large component

Reply:
We talked about this: 
One option could be to separate the different parts of the component in different files like explained here https://vuejs.org/v2/guide/single-file-components.html#What-About-Separation-of-Concerns 
or move parts of the template into further sub components. 
Another idea is to factor out the `TerminalSession` class but it could even make it harder to read and maintain the code as it is interwoven with the component (needs access to `vm`).
For now we agreed to leave it as is.